chrome/content/zotero/overlay.js

```diff
@@ -209,7 +210,14 @@ var ZoteroOverlay = new function()
 			
 			// Make visible
 			ZoteroPane.makeVisible();
-			
+
+			// Warn about unsafe data directory on first display
+			if(Zotero.Prefs.get('warnUnsafeDataDirOnStartup') && this.warnUnsafeDataDir) {
```

Comment:
These checks should just go in Zotero.checkDataDirectory(), with the flag set on the `Zotero` object, and this code should just call Zotero.checkDataDirectory(). (Function overhead doesn't matter.) There's only one `Zotero`, but there can be multiple `ZoteroOverlay`s, so this will reappear if a second window was opened.